### PR TITLE
fix: styles not working as expected in plugin

### PR DESCRIPTION
**Pull request by KaWaite**
# Overview
When you have a complex plugin that uses external stylesheets, for example antd + styled components, styles weren't showing properly.
This was because we set the plugin's html to the doc.body.innerHTML. For stylesheets, the order of elements is important, so this moves all the link elements out of the body and in to the head where they belong.

### CODE REVIEW

File: src/components/atoms/Plugin/IFrame/hooks.ts [hunk 1]
```diff
@@ -131,6 +131,11 @@ export default function useHook({
 
     doc.body.innerHTML = html;
 
+    const links = doc.body.querySelectorAll("link");
+    if (links) {
+      linksToHead(doc, links);
+    }
```
Comment (rot1024) [inline src/components/atoms/Plugin/IFrame/hooks.ts:137]:
```suggestion
    linksToHead(doc.body.querySelectorAll("link"));
```

File: src/components/atoms/Plugin/IFrame/hooks.ts [hunk 2]
```diff
@@ -228,3 +233,14 @@ function runScript(oldScript: HTMLScriptElement) {
     }
   });
 }
+
+function linksToHead(doc: Document, links: NodeListOf<HTMLLinkElement>) {
+  for (const link of links) {
+    const newLink = document.createElement("link");
+    newLink.rel = link.rel;
+    newLink.href = link.href;
+    newLink.appendChild(document.createTextNode(link.innerText));
```
Comment (rot1024) [inline src/components/atoms/Plugin/IFrame/hooks.ts:242]:
```suggestion
    if (link.innerText) {
      newLink.appendChild(document.createTextNode(link.innerText));
    }
```

File: src/components/atoms/Plugin/IFrame/hooks.ts [hunk 3]
```diff
@@ -228,3 +233,14 @@ function runScript(oldScript: HTMLScriptElement) {
     }
   });
 }
+
+function linksToHead(doc: Document, links: NodeListOf<HTMLLinkElement>) {
+  for (const link of links) {
+    const newLink = document.createElement("link");
+    newLink.rel = link.rel;
+    newLink.href = link.href;
```
Comment (rot1024) [inline src/components/atoms/Plugin/IFrame/hooks.ts:241]:
```suggestion
    for (const attr of newLink.attributes) {
      newLink.setAttribute(attr.name, attr.value);
    }
```

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: src/components/atoms/Plugin/IFrame/hooks.ts (hunk 3 above)
Comment (KaWaite) [reply src/components/atoms/Plugin/IFrame/hooks.ts:241]: do you mean `for (const attr of link.attributes)` ? Not newLink?

File: src/components/atoms/Plugin/IFrame/hooks.ts (hunk 2 above)
Comment (KaWaite) [reply src/components/atoms/Plugin/IFrame/hooks.ts]: Updated [here](https://github.com/reearth/reearth-web/pull/322/commits/b1879c0e926dfa34b636a872abfa7a078e4d8615)

File: src/components/atoms/Plugin/IFrame/hooks.ts (hunk 1 above)
Comment (KaWaite) [reply src/components/atoms/Plugin/IFrame/hooks.ts]: Updated [here](https://github.com/reearth/reearth-web/pull/322/commits/b1879c0e926dfa34b636a872abfa7a078e4d8615)

File: src/components/atoms/Plugin/IFrame/hooks.ts (hunk 3 above)
Comment (KaWaite) [reply src/components/atoms/Plugin/IFrame/hooks.ts]: Might be updated [here](https://github.com/reearth/reearth-web/pull/322/commits/b1879c0e926dfa34b636a872abfa7a078e4d8615)

**Review by rot1024 — APPROVED** [review]

**Pull request merged**